pkg/client/devops/jenkins/pure_request.go

```diff
@@ -47,13 +47,27 @@ func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *dev
 	apiURL.RawQuery = httpParameters.Url.RawQuery
 	client := &http.Client{Timeout: 30 * time.Second}
 
-	header := httpParameters.Header
+	header := httpParameters.Header.Clone()
```

Comment:
It will panic if the header is nil. If the header never be nil, then you don't need to check it `if header == nil {`

Reply:
Please note that nil Is a valid method receiver value, see below sample code:

```go
package main

import (
	"fmt"
)

type Header map[string][]string

// Clone returns a copy of h or nil if h is nil.
func (h Header) Clone() Header {
	if h == nil {
		return nil
	}

	// Find total number of values.
	nv := 0
	for _, vv := range h {
		nv += len(vv)
	}
	sv := make([]string, nv) // shared backing array for headers' values
	h2 := make(Header, len(h))
	for k, vv := range h {
		n := copy(sv, vv)
		h2[k] = sv[:n:n]
		sv = sv[n:]
	}
	return h2
}

func main() {
	fmt.Println("Hello, playground")
	var header Header = nil
	header.Clone()
}
```

Reply:
Learned from it. But I still feel confused.